File: app/src/main/java/com/MaxEle/maximarius/nir_navigation/util/BillingClientSetup.java
package com.MaxEle.maximarius.nir_navigation.util;

import android.content.Context;

import com.android.billingclient.api.BillingClient;
import com.android.billingclient.api.PurchasesUpdatedListener;

public class BillingClientSetup {
    private static BillingClient instance;

    public static BillingClient getInstance(Context context, PurchasesUpdatedListener listener) {
        return instance == null ? setupBillingClient(context,listener):instance;
    }

    private static BillingClient setupBillingClient(Context context, PurchasesUpdatedListener listener) {
        return BillingClient.newBuilder(context)
                .enablePendingPurchases()
                .setListener(listener)
                .build();
    }
}

File: app/src/main/java/com/MaxEle/maximarius/nir_navigation/util/BillingClientSetup.kt
package com.MaxEle.maximarius.nir_navigation.util

import android.content.Context
import com.android.billingclient.api.BillingClient
import com.android.billingclient.api.PurchasesUpdatedListener
import com.MaxEle.maximarius.nir_navigation.util.BillingClientSetup

object BillingClientSetup {
    private val instance: BillingClient? = null
    @JvmStatic
    fun getInstance(context: Context, listener: PurchasesUpdatedListener): BillingClient {
        return instance ?: setupBillingClient(context, listener)
    }

    private fun setupBillingClient(
        context: Context,
        listener: PurchasesUpdatedListener
    ): BillingClient {
        return BillingClient.newBuilder(context)
            .enablePendingPurchases()
            .setListener(listener)
            .build()
    }
}